### ISSUE TITLE
Floats in configuration probably shouldn't round to integers transparently

### ISSUE THREAD

**Issue by nagisa** [mehcode/config-rs#160]
Currently when you `Config::get` an integer and config has a float specified, the `config-rs` crate will transparently lose the precision, potentially leading to unexpected behaviour. For instance:

```toml
frame_rate = 59.94
```

will transparently round up to 60, potentially giving wrong video. There are probably many more instances where loss of precision can give very invalid results.

I believe the relevant code is probably [this line](https://github.com/mehcode/config-rs/blob/bf09a730d44cb72f483dab04163dd75f2b9d1746/src/value.rs#L223).

I saw a number of `FIXME`s in the codebase around not succeeding when [integer wraparound occurs](https://github.com/mehcode/config-rs/blob/bf09a730d44cb72f483dab04163dd75f2b9d1746/src/de.rs#L76). We most likely can utilize something similar here – if converting results in loss of precision, fail.

**Comment by matthiasbeyer** [comment]
Agreed. Also, the crate stores u8 in u64 (and so on), which should be fixed as well (mentioning it here because it falls into the same category of issues).
